### PR TITLE
Cleaned up Rustproof with a linter

### PR DESCRIPTION
**Pull request by bajr**
Some code is now more closely representative of idiomatic rust. Did not include expression_parser.rs or removal of `return` statements. I tested and confirmed this code runs the way it is expected to with respect to these changes.

### CODE REVIEW

**Comment by salterm** [comment]
In smt_output/mod.rs line ~144, why does the LessThan match arm generate it's left and right operands a second time?

In expression/mod.rs line ~160, why does our example of String output use single quotes instead of double quotes? ("bool" -> 'bool')

Also in expression/mod.rs line 33, you added a TODO to "Fix these enum variant names not to end with the enum name". What is the rationale for this, and what it is talking about?

File: src/smt_output/mod.rs [hunk 1]
```diff
@@ -134,46 +134,49 @@ impl Pred2SMT for SMTLib2<QF_ABV> {
...
+                        if determine_evaluation_type(vc).starts_with('i') {
                             return self.assert(bitvec::OpCodes::BvAShr, &[l,r]);
                         } else {
                             return self.assert(bitvec::OpCodes::BvLShr, &[l,r]);
                         }
                     },
                     BinaryOperator::LessThan => {
-                        if determine_evaluation_type(b.left.as_ref()).starts_with("i") {
+                        let l = self.expr2smtlib(b.left.as_ref());
+                        let r = self.expr2smtlib(b.right.as_ref());
```
Comment (bajr) [inline src/smt_output/mod.rs]: Looks like I missed these when I was resolving merge conflicts on my local branch. I'll have them fixed soon.

File: src/weakest_precondition/mod.rs [hunk 2]
```diff
@@ -160,11 +157,11 @@ pub fn gen(index: usize, data: &mut MirData, post_expr: &Option<Expression>, deb
     wp
 }
 
-/// Returns the type of an operand as a String (ie: "i32", "bool", etc.)
+/// Returns the type of an operand as a `String` (ie: `i32`, `bool`, etc.)
```
Comment (bajr) [inline src/weakest_precondition/mod.rs]: @salterm If this is the line you're referring to, these aren't single quotes, but backticks, which formats this word to be monospaced in the documentation.

File: src/expression/mod.rs [hunk 3]
```diff
@@ -30,6 +30,7 @@ pub struct UnaryExpressionData {
     pub e: Box<Expression>
 }
 
+// TODO Fix these enum variant names not to end with the enum name
```
Comment (bajr) [inline src/expression/mod.rs]: I was hesitant to outright fix the warning message generated for this one from the linter, given how much we use these names in the code. However, I think the linter's reasoning is valid. 

In short, it's too redundant and wordy to have an enum named `Expression`, and then to also have its variants include that same name in their names (i.e. `BinaryExpression`) when it's clear they're already some variety of the enum. I'm less sure of what to rename them to, unfortunately, but perhaps just `Binary` would be sufficient.

Here's a link to the linter warning: https://github.com/Manishearth/rust-clippy/wiki#enum_variant_names

File: src/weakest_precondition/mod.rs (hunk 2 above)
Comment (bajr) [inline src/weakest_precondition/mod.rs]: Oh wait, I see what you mean... I'll fix this, too.

**Comment by salterm** [comment]
@bajr I would be amenable to a change in the names for Expression enum variants; my concern was mostly with the cryptic // TODO line, especially since we have this nice expression tracker on GitHub.

**Pull request merged**